csdmpy/csdm.py

```diff
@@ -753,6 +732,30 @@ def ndim(self):
         """Return the total number of dimensions."""
         return len(self.dimensions)
 
+    def reshape(self, shape):
+        """Reshape the csdm object to shape.
+
+        Args:
+            shape: A list of dimension objects or integers.
+        """
+        size = self.y[0].components[0].size
+
+        shape_int = [item if isinstance(item, int) else item.size for item in shape]
+        new_dim = []
+        for index in shape:
+            sh = int(-size / np.prod(shape_int)) if index == -1 else index
```

Comment:
I'm having difficulty understanding why this line is checking for `index ==  -1`. Wouldn't `index` always be an instance of a positive `int` or `csdmpy.Dimension` object?